### PR TITLE
CrowdStrike Falcon Intel v2

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24295

## Description
CrowdStrike Falcon Intel v2

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by guyfreund** [comment]
@Arsenikr Please go over the yml and fill in old missing description, you can search for `TBD` and you will find them all.

**Comment by Arsenikr** [comment]
> @Arsenikr Please go over the yml and fill in all missing descriptions, you can search for `TBD` and you will find them all.

@guyfreund done

**Comment by guyfreund** [comment]
@kirbles19 yml and docs are ready to be reviewed.
@Arsenikr is there any secret data in the docs we want to remove?

**Review by Arsenikr — APPROVED** [review]

File: Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/CrowdStrikeFalconIntel_v2.py [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+    if isinstance(keys, str):
+        new_list = [item.get(keys) for item in items_list]
+    elif isinstance(keys, list):
+        new_list = [{underscoreToCamelCase(f): item.get(f) for f in item if f in keys} for item in items_list]
+    if return_type == 'list':
+        return new_list
+    return ', '.join(str(item) for item in new_list)
+
+
+def get_indicator_outputs(o: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (ShahafBenYakir) [inline Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/CrowdStrikeFalconIntel_v2.py:272]: Replace `o` with a more meaningful name

**Review by ShahafBenYakir — COMMENTED** [review]
Don't we deprecate the old integration?

File: Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/README.md [hunk 2]
```diff
@@ -0,0 +1,947 @@
...
+This integration was integrated and tested with version xx of CrowdStrike Falcon Intel v2
+## Configure CrowdStrike Falcon Intel v2 on Cortex XSOAR
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for CrowdStrike Falcon Intel v2.
+3. Click **Add instance** to create and configure a new integration instance.
+
+| **Parameter** | **Description** | **Required** |
+| --- | --- | --- |
+| credentials | Client ID | True |
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/README.md:11]: Falcon X uses the same parameter, what is wrong with it?
image

File: Packs/ApiModules/Scripts/CrowdStrikeApiModule/README.md [hunk 3]
```diff
@@ -11,3 +11,5 @@ from CrowdStrikeApiModule import *  # noqa: E402
 if __name__ in ["builtins", "__main__"]:
     main()
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Packs/ApiModules/Scripts/CrowdStrikeApiModule/README.md:12]: what?

[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Packs/ApiModules/Scripts/CrowdStrikeApiModule/README.md:12]: I don't understand this comment/description...

File: Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Packs/CrowdStrikeIntel/Integrations/CrowdStrikeFalconIntel_v2/README.md:11]: I mean that the description is bad. Credentials != Client ID... Credentials are Client ID and?

**Comment by ShahafBenYakir** [comment]
Don't we deprecate the old integration?

File: Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py [hunk 4]
```diff
@@ -31,6 +30,14 @@ def _error_handler(res: requests.Response):
             errors = error_entry.get('errors', [])
             err_msg += '\n'.join(f"{error.get('code')}: {error.get('message')}" for  # pylint: disable=no-member
                                  error in errors)
+            if 'Failed to issue access token - Not Authorized' in err_msg:
+                err_msg = err_msg.replace('Failed to issue access token - Not Authorized',
+                                          f'Client Secret ({self._client_secret}) is invalid.')
```
Comment (ShahafBenYakir) [inline Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py:35]: Not secure.
```suggestion
                                          f'Client Secret is invalid.')
```

**Comment by ShahafBenYakir** [comment]
1. Change the client ID to credentials.
2. Report `sort` parms should be separated into field and order.
3. Change report param - `q` to a more meaningful name (free_search?).
4. Add link to docs for the `query`  param

**Review by ShahafBenYakir — APPROVED** [review]

File: Packs/ApiModules/ReleaseNotes/1_1_4.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+#### Scripts
+##### CrowdStrikeApiModule
+- Improved error handling.
+- Update docker image from `3.8.3.9324` to `3.8.5.10845`.
```
Comment (amshamah419) [inline Packs/ApiModules/ReleaseNotes/1_1_4.md:5]: ```- Updated docker image from `3.8.3.9324` to `3.8.5.10845`.```

**Review by amshamah419 — COMMENTED** [review]
Just make the changes and you're good to go.

**Pull request merged**